### DISCUSSION TITLE
Support for scientific notation in table output

### DISCUSSION THREAD

**Discussion by maxpaton** [kdbinsidebrains/plugin#91] [General]
Hi there,

Thanks a lot for your plugin, I like it a lot.

Is there any chance you could add support for scientific notation in the 'Table Result' output? For numbers like 0.00000000000001 it would be nice if you could see that as 1e-14, instead of having to turn the float precision all the way up to avoid the number being shown as 0.

Thanks a lot!
Max

**Comment by smklimenko** [comment]
Hi Max,

Yes, I think that's possible as an option in configs by default and as a thousands separator in Table View options: https://www.kdbinsidebrains.dev/features/tables#thousands-separator

Ill check the best way here.

**Reply by maxpaton** [reply]
Hi,

I'm afraid it doesn't look like it does (see screenshot below). Here the zeros correspond to the KDB float 0.0000000000001, which have showed up as zero despite be enabling the thousands separator. Unless I am missing something?

![image](https://github.com/kdbinsidebrains/plugin/assets/40034159/5fb60418-2160-451f-9bd8-55b20193d516)

Thanks

**Reply by smklimenko** [reply]
it doesn't. I just said it will be implemented the same way once implemented

**Reply by maxpaton** [reply]
Sorry I misunderstood! Okay great, I'll wait to hear from you on the confirmation of implementation.

Thanks a lot.

**Comment by smklimenko** [comment]
Try version 5.4.0: https://plugins.jetbrains.com/plugin/16746-kdbinsidebrains/edit/versions/stable/491129

It's disabled by default for now, so you have to enabled it in Settings: https://www.kdbinsidebrains.dev/settings/options#decimal-format or each time in the Table View: https://www.kdbinsidebrains.dev/features/tables#scientific-notation

**Reply by maxpaton** [reply]
Great, thanks a lot for doing that!
